apps/web/src/components/AssessmentResultsTable/utils.tsx

```diff
@@ -0,0 +1,260 @@
+
+import {
+  AssessmentDecision,
+  AssessmentResult,
+  AssessmentResultType,
+  AssessmentStep,
+  AssessmentStepType,
+  PoolSkillType,
+} from "@gc-digital-talent/graphql";
+import {
+  getAssessmentDecision,
+  getAssessmentStepType,
+} from "@gc-digital-talent/i18n/src/messages/localizedConstants";
+import { commonMessages } from "@gc-digital-talent/i18n";
+
+import poolCandidateMessages from "~/messages/poolCandidateMessages";
+
+import {
+  AssessmentTableRow,
+  AssessmentTableRowColumn,
+  AssessmentTableRowColumnProps,
+  ColumnStatus,
+  StatusColor,
+} from "./types";
+import cells from "../Table/cells";
+import Dialog from "../ScreeningDecisions/ScreeningDecisionDialog";
+
+const iconColorMap: Record<StatusColor, Record<string, string>> = {
+  error: {
+    "data-h2-color": "base(error)",
+  },
+  hold: {
+    "data-h2-color": "base(secondary)",
+  },
+  toAssess: {
+    "data-h2-color": "base(quaternary)",
+  },
+  success: {
+    "data-h2-color": "base(success)",
+  },
+  gray: {
+    "data-h2-color": "base(gray)",
+  },
+};
+
+export const columnHeader = (
+  header: string,
+  status: ColumnStatus,
+  type: AssessmentStepType | null,
+  intl: IntlShape,
+) => {
+  const Icon = status.icon;
+  let ariaLabel;
+  switch (status.color) {
+    case "error":
+      ariaLabel =
+        type === AssessmentStepType.ApplicationScreening ||
+        type === AssessmentStepType.ScreeningQuestionsAtApplication
+          ? intl.formatMessage(poolCandidateMessages.screenedOut)
+          : intl.formatMessage(poolCandidateMessages.unsuccessful);
+      break;
+    case "hold":
+      ariaLabel = intl.formatMessage(poolCandidateMessages.holdForAssessment);
+      break;
+    case "toAssess":
+      ariaLabel = intl.formatMessage(poolCandidateMessages.toAssess);
+      break;
+    case "success":
+      ariaLabel =
+        type === AssessmentStepType.ApplicationScreening ||
+        type === AssessmentStepType.ScreeningQuestionsAtApplication
+          ? intl.formatMessage(poolCandidateMessages.screenedIn)
+          : intl.formatMessage(poolCandidateMessages.successful);
+      break;
+    default:
+      ariaLabel = intl.formatMessage(commonMessages.notApplicable);
+      break;
+  }
+
+  return (
+    <span data-h2-display="base(inline)">
+      <span
+        data-h2-display="base(flex)"
+        data-h2-align-content="base(center)"
+        data-h2-gap="base(x.25)"
+      >
+        {Icon && (
+          <Icon
+            aria-label={ariaLabel}
+            aria-hidden="false"
+            {...iconColorMap[status.color]}
+            data-h2-display="p-tablet(inline-block)"
+            data-h2-vertical-align="base(middle)"
+            data-h2-height="base(auto)"
+            data-h2-flex-shrink="base(0)"
+            data-h2-width="base(x.85)"
+          />
+        )}
+        {header}
+      </span>
+    </span>
+  );
+};
+
+export const columnStatus = (
```

Comment:
Doh! There'a lot of overlap here with utils/poolCandidate.ts. Sorry for not pointing that out sooner. Lets try to reuse the util functions, to avoid maintaining in two places. I recommend using `getResultsDecision` and then simply mapping the decisions to ComnStatus.

Reply:
Haha ya tons of overlap however the `getResultsDecision` is much cleaner. Fixed by [ac2a1e2](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9324/commits/ac2a1e23af0dc7e74c79a43350e1dbc2f17557c4)